packages/frameworks-solid-start/src/handler.ts

```diff
@@ -0,0 +1,132 @@
+import { AuthHandler, type AuthAction, type AuthOptions } from "@auth/core";
+import { Cookie, parseString, splitCookiesString } from "set-cookie-parser";
+import { serialize } from "cookie";
+
+export interface SoliduthOptions extends AuthOptions {
+  /**
+   * Defines the base path for the auth routes.
+   * @default '/api/auth'
+   */
+  prefix?: string;
+}
+
+const actions: AuthAction[] = [
+  "providers",
+  "session",
+  "csrf",
+  "signin",
+  "signout",
+  "callback",
+  "verify-request",
+  "error",
+  "_log",
+];
+
+export type ISolidAuthHandlerOpts = AuthOptions & {
+  failureRedirect?: string;
+};
+
+// currently multiple cookies are not supported, so we keep the next-auth.pkce.code_verifier cookie for now:
+// because it gets updated anyways
+// src: https://github.com/solidjs/solid-start/issues/293
+const getSetCookieCallback = (cook?: string | null): Cookie | undefined => {
+  if (!cook) return;
+  const splitCookie = splitCookiesString(cook);
+  for (const cookName of [
+    "__Secure-next-auth.session-token",
+    "next-auth.session-token",
+    "next-auth.pkce.code_verifier",
+    "__Secure-next-auth.pkce.code_verifier",
```

Comment:
Relying on hardcoded cookie names is not feasible. This totally bypasses the default behavior (eg.: [dynamic prefixing](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/cookie.ts#L58), [user config preference](https://github.com/nextauthjs/next-auth/blob/2e924edcdfae984978d9a184768706381d7d624f/packages/core/src/lib/init.ts#L75-L81), [session chunking](https://github.com/nextauthjs/next-auth/blob/main/packages/core/src/lib/cookie.ts#L129), etc.)


Since this is experimental, it's fine to push out to have something working, but we will have to eventually remove this.

Reply:
ye i agree but what can we do? its an actual solid bug https://github.com/solidjs/solid-start/issues/293
for now it is working on both prod and ev so i will push it, if something changes in the core i will make sure to update it aswell.